tools/crm_mon.c

```diff
@@ -802,8 +831,71 @@ main(int argc, char **argv)
         crm_bump_log_level(argc, argv);
     }
 
+    /* Which output format to use could come from two places:  The --as-xml
+     * style arguments we gave in mode_entries above, or the formatted output
+     * arguments added by pcmk__register_formats.  If the latter were used,
+     * output_format will be mon_output_unset.
+     *
+     * Call the callbacks as if those older style arguments were provided so
+     * the various things they do get done.
+     */
+    if (output_format == mon_output_unset) {
+        /* NOTE:  There is no way to specify CGI mode or simple mode with --output-as.
+         * Those will need to get handled eventually, at which point something else
+         * will need to be added to this block.
+         */
+        if (safe_str_eq(args->output_ty, "html")) {
+            as_html_cb("h", args->output_dest, NULL, &error);
+        } else if (safe_str_eq(args->output_ty, "text")) {
+            no_curses_cb("N", NULL, NULL, &error);
+        } else if (safe_str_eq(args->output_ty, "xml")) {
+            as_xml_cb("X", args->output_dest, NULL, &error);
+        } else {
+            /* Neither old nor new arguments were given, so set the default. */
+            if (args->output_ty != NULL) {
+                free(args->output_ty);
+            }
+
+            args->output_ty = strdup("console");
+            output_format = mon_output_console;
+        }
+    }
+
+    rc = pcmk__output_new(&out, args->output_ty, args->output_dest, argv);
+    if (rc != 0) {
+        fprintf(stderr, "Error creating output format %s: %s\n", args->output_ty, pcmk_strerror(rc));
+        return clean_up(CRM_EX_ERROR);
+    }
+
+    /* If certain format options were specified, we want to set some extra
+     * options.  We can just process these like they were given on the
+     * command line.
+     */
+    if (output_format == mon_output_plain) {
+        gchar **extra_args = g_strsplit("--fancy 1", " ", -1);
+        g_option_context_parse_strv(context, &extra_args, &error);
+        g_strfreev(extra_args);
```

Comment:
I could see this being useful elsewhere, maybe we should functionize it? Something like force_args(context, &error, FMT, ...)

Reply:
It is likely to be useful in other command line tools.  And it's also pretty repetitive here.  Moving it into lib/common/cmdline.c looks like a good idea.